tests/integration/vlan_test.py

```diff
@@ -234,3 +236,87 @@ def create_two_vlans_state():
             },
         ]
     }
+
+
+@pytest.mark.xfail(reason="https://bugzilla.redhat.com/1722352")
+def test_change_vlan_id(eth1_up):
+    vlana_name = "vlan.a"
+    vlanb_name = "vlan.b"
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: vlana_name,
+                Interface.TYPE: InterfaceType.VLAN,
+                Interface.STATE: InterfaceState.UP,
+                Interface.MTU: 1280,
+                Interface.MAC: "d4:ee:07:25:42:5a",
+                VLAN.CONFIG_SUBTREE: {VLAN.ID: 201, VLAN.BASE_IFACE: "eth1"},
+                Interface.IPV4: {
+                    InterfaceIPv4.ENABLED: True,
+                    InterfaceIPv4.ADDRESS: [
+                        {
+                            InterfaceIPv4.ADDRESS_IP: "192.0.2.251",
+                            InterfaceIPv4.ADDRESS_PREFIX_LENGTH: 24,
+                        }
+                    ],
+                },
+                Interface.IPV6: {
+                    InterfaceIPv6.ENABLED: True,
+                    InterfaceIPv6.ADDRESS: [
+                        {
+                            InterfaceIPv6.ADDRESS_IP: "2001:db8:1::1",
+                            InterfaceIPv6.ADDRESS_PREFIX_LENGTH: 64,
+                        }
+                    ],
+                },
+            },
+            {
+                Interface.NAME: vlanb_name,
+                Interface.TYPE: InterfaceType.VLAN,
+                Interface.STATE: InterfaceState.UP,
+                Interface.MTU: 1280,
+                VLAN.CONFIG_SUBTREE: {VLAN.ID: 202, VLAN.BASE_IFACE: "eth1"},
+                Interface.IPV4: {
+                    InterfaceIPv4.ENABLED: True,
+                    InterfaceIPv4.DHCP: True,
+                    InterfaceIPv4.AUTO_DNS: False,
+                    InterfaceIPv4.AUTO_GATEWAY: False,
+                    InterfaceIPv4.AUTO_ROUTES: False,
+                    InterfaceIPv4.AUTO_ROUTE_TABLE_ID: 100,
+                },
+                Interface.IPV6: {
+                    InterfaceIPv6.ENABLED: True,
+                    InterfaceIPv6.DHCP: True,
+                    InterfaceIPv6.AUTOCONF: True,
+                    InterfaceIPv6.AUTO_DNS: True,
+                    InterfaceIPv6.AUTO_GATEWAY: True,
+                    InterfaceIPv6.AUTO_ROUTES: True,
+                    InterfaceIPv6.AUTO_ROUTE_TABLE_ID: 100,
+                },
+            },
+        ]
+    }
```

Comment:
Instead of using this, I recommend you to do:
```
def test_change_vlan_id(vlan_on_eth1):
  vlan_state = vlan_on_eth1[Interface.KEY][1]
  # Here you can modify the state (change vlan id)
```

Reply:
IIUC, vlan_on_eth1 is hard coded using VLAN_IFNAME = "eth1.101", right? But in my case, a name like "eth1.$vlan_id" cannot work: bz1902976. So... days ago I asked you about using global variable to change VLAN_IFNAME :)
If vlan_on_eth1 is really restricted, I think I can use vlan_interface("vlan.a", 101, eth1_up[Interface.KEY][0][Interface.NAME]) ?